internal/executor/service/job_requester.go

```diff
@@ -2,99 +2,196 @@ package service
+	executorContext "github.com/armadaproject/armada/internal/executor/context"
+	"github.com/armadaproject/armada/internal/executor/job"
+	"github.com/armadaproject/armada/internal/executor/reporter"
+	"github.com/armadaproject/armada/internal/executor/util"
+	"github.com/armadaproject/armada/internal/executor/utilisation"
 	"github.com/armadaproject/armada/pkg/api"
 	"github.com/armadaproject/armada/pkg/armadaevents"
 	"github.com/armadaproject/armada/pkg/executorapi"
 )
 
-type LeaseRequester interface {
-	LeaseJobRuns(availableResource *armadaresource.ComputeResources, nodes []*api.NodeInfo, unassignedJobRunIds []armadaevents.Uuid) ([]*executorapi.JobRunLease, []*armadaevents.Uuid, error)
+type JobRequester struct {
+	leaseRequester     LeaseRequester
+	eventReporter      reporter.EventReporter
+	utilisationService utilisation.UtilisationService
+	clusterId          executorContext.ClusterIdentity
+	podDefaults        *configuration.PodDefaults
+	jobRunStateStore   job.RunStateStore
 }
 
-type JobLeaseRequester struct {
-	executorApiClient executorapi.ExecutorApiClient
-	clusterIdentity   clusterContext.ClusterIdentity
-	minimumJobSize    armadaresource.ComputeResources
-}
-
-func NewJobLeaseRequester(
-	executorApiClient executorapi.ExecutorApiClient,
-	clusterIdentity clusterContext.ClusterIdentity,
-	minimumJobSize armadaresource.ComputeResources,
-) *JobLeaseRequester {
-	return &JobLeaseRequester{
-		executorApiClient: executorApiClient,
-		clusterIdentity:   clusterIdentity,
-		minimumJobSize:    minimumJobSize,
+func NewJobRequester(
+	clusterId executorContext.ClusterIdentity,
+	eventReporter reporter.EventReporter,
+	leaseRequester LeaseRequester,
+	jobRunStateStore job.RunStateStore,
+	utilisationService utilisation.UtilisationService,
+	podDefaults *configuration.PodDefaults,
+) *JobRequester {
+	return &JobRequester{
+		leaseRequester:     leaseRequester,
+		eventReporter:      eventReporter,
+		utilisationService: utilisationService,
+		jobRunStateStore:   jobRunStateStore,
+		clusterId:          clusterId,
+		podDefaults:        podDefaults,
 	}
 }
 
-func (requester *JobLeaseRequester) LeaseJobRuns(
-	availableResource *armadaresource.ComputeResources,
-	nodes []*api.NodeInfo,
-	unassignedJobRunIds []armadaevents.Uuid,
-) ([]*executorapi.JobRunLease, []*armadaevents.Uuid, error) {
+func (r *JobRequester) RequestJobsRuns() {
+	leaseRequest, err := r.createLeaseRequest()
+	if err != nil {
+		log.Errorf("Failed to create lease request because %s", err)
+		return
+	}
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
-	stream, err := requester.executorApiClient.LeaseJobRuns(ctx, grpcretry.Disable(), grpc.UseCompressor(gzip.Name))
+	leaseResponse, err := r.leaseRequester.LeaseJobRuns(ctx, leaseRequest)
+	if err != nil {
+		log.Errorf("Failed to request new jobs leases as because %s", err)
+		return
+	}
+	logAvailableResources(leaseRequest.AvailableResource, len(leaseResponse.LeasedRuns))
```

Comment:
I'd log  a summary here of what you got- e.g. how many new jobs, how many jobs to preempt etc.